### PR TITLE
Editorial: Use SVG for Generator Objects Relationships figure

### PR DESCRIPTION
**Pull request by arai-a**
Derived from https://github.com/tc39/ecma262/pull/3238

Given the rendering result of `uxf` is highly environment dependent, switch the "Generator Objects Relationships" figure image to a SVG file, ~~and a PNG file converted from the SVG file~~, in order to make the further update easy.

The content follows the current PNG file's style, with slight modification to the position/size and font size, to make it simpler.

I'll modify the file again in https://github.com/tc39/ecma262/pull/3238 to reflect the update there.

### CODE REVIEW

**Comment by ljharb** [comment]
Do we even need the png, given that svgs can display natively?

**Comment by arai-a** [comment]
Good point
given that module-related figure uses only SVG,  I'll drop `figure-2.png`.

File: spec.html [hunk 1]
```diff
@@ -47259,7 +47259,7 @@ <h1>Properties of Promise Instances</h1>
     <h1>GeneratorFunction Objects</h1>
     <p>GeneratorFunctions are functions that are usually created by evaluating |GeneratorDeclaration|s, |GeneratorExpression|s, and |GeneratorMethod|s. They may also be created by calling the %GeneratorFunction% intrinsic.</p>
     <emu-figure id="figure-2" caption="Generator Objects Relationships" informative>
-      <img alt="A staggering variety of boxes and arrows." src="img/figure-2.png">
+      <img alt="A staggering variety of boxes and arrows." height="660" src="img/figure-2.svg" width="900">
```
Comment (ljharb) [inline spec.html:47262]: i think the width/height info can be inside the svg data, no?

Comment (arai-a) [reply spec.html:47262]: The SVG file also contains those info, but having the `width`/`height` attributes helps the layout until the SVG gets loaded.
Other `` elements also specify those attributes.

**Comment by michaelficarra** [comment]
The SVG LGTM, though if we're embedding the SVG directly in the document, we should be consistent and do the same for `img/figure-1.svg` (and delete its associated raster).

**Review by michaelficarra — APPROVED** [review]

**Review by syg — COMMENTED** [review]
Great work converting this outdated thing to SVG! However I see overflowing text in the SVG rendering on my Chromebook. Could you choose a particular font, perhaps? The Chromebook's list of fonts is attached in the screenshot.
Screenshot 2023-12-13 15 29 26

**Comment by arai-a** [comment]
I've checked the rendering with Arimo (which would match with `sans-serif`), but I don't see the overflow on macOS + Firefox/Chromium/Safari.
@syg, can you post a screenshot?

here's the rendering with Chromium with Arimo for me
chromium-render-with-arimo

**Comment by syg** [comment]
On my Chromebook: 
Screenshot 2023-12-13 17 22 37

118.0.5993.124 (Official Build) (64-bit) 

Maybe it's a ChromeOS thing.

**Comment by arai-a** [comment]
I've made boxes slightly wider for those overflowing cases.
does this look good?

**Comment by syg** [comment]
Thanks for adjusting. It's better, but still not entirely within the box.
Screenshot 2023-12-18 11 59 57

**Comment by arai-a** [comment]
Okay, so here's the current situation:
  * The text on the figure uses `font-size: 14px` in most case
  * `%GeneratorFunction.prototype%` and `%GeneratorFunction.prototype.prototype%` uses `font-size: 13px` because it's too long for the box, and reducing the font size makes the width of the text smaller and it fits the box
  * On the Chromebook environment, `font-size: 13px` makes the font size smaller than `font-size: 14px`, but the width taken by the entire text is same (so, only each glyph gets smaller, without changing the spacing?)

So, better fix would be to modify the layout (instead of keeping the original PNG's layout) and use `font-size: 14px` consistently, while making the box wide enough.

**Comment by michaelficarra** [comment]
> use `font-size: 14px` consistently

That WFM.

> * the width taken by the entire text is same (so, only each glyph gets smaller, without changing the spacing?)

This is really strange behaviour. It might be worth reporting, but I'm not sure to whom. Blink?

**Comment by syg** [comment]
> This is really strange behaviour. It might be worth reporting, but I'm not sure to whom. Blink?

Good question, but it's likely ChromeOS and not blink, because Chrome on non-ChromeOS is fine here.

**Comment by syg** [comment]
The new copy looks great, thanks.

**Review by bakkot — APPROVED** [review]

**Review by syg — APPROVED** [review]

**Pull request merged**